src/ResourceManager/Insights/Commands.Insights/Diagnostics/RemoveAzureRmDiagnosticSettingCommand.cs

```diff
@@ -0,0 +1,149 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.Monitor;
+using Microsoft.Azure.Management.Monitor.Models;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+using System.Globalization;
+using System.Management.Automation;
+using System.Net;
+using System.Threading;
+
+namespace Microsoft.Azure.Commands.Insights.Diagnostics
+{
+    /// <summary>
+    /// Removes a named diagnostic setting or disables the setting called 'service' if the name argument is not present or if is 'service'.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDiagnosticSetting", SupportsShouldProcess = true), OutputType(typeof(AzureOperationResponse))]
+    public class RemoveAzureRmDiagnosticSettingCommand : ManagementCmdletBase
+    {
+        /// <summary>
+        /// This is a temporary constant to provide backwards compatibility
+        /// </summary>
+        internal const string TempServiceName = "service";
+
+        #region Parameters declarations
+
+        /// <summary>
+        /// Gets or sets the resourceId parameter of the cmdlet
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
```

Comment:
@pkamath13 @gucalder does the SDK only accept a resource id? Is the user not able to provide the components of the id to the server? It might be worth looking into constructing the resource id for the user if all they would need to provide is a resource group name and name of the resource.

Reply:
This resource name can be virtually any resource that can be monitored since the diagnostic settings are identified this way: {resourceId}/providers/microsoft.insights/diagnosticSettings/{settingName}. SettingName can be "service" or any other name given by the user.  So we will need each component of the resource given to us: subId, rg name, resource type, resource name.
Also, the Set cmdlet also receives resourceId.

Can I consider this as a new parameter set in future iterations?